include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -522,6 +522,141 @@ class cpp_dec_float
       ar& boost::make_nvp("precision", prec_elem);
    }
 
+   void eval_round_self()
+   {
+      const bool needs_rounding = (((isfinite)() == true) && (iszero() == false));
+
+      if(needs_rounding)
+      {
+         using local_size_type = typename array_type::size_type;
+         using local_limb_type = typename array_type::value_type;
+
+         std::int32_t digits_limb_0 = 0;
+
+         local_limb_type tmp_limb_0 = data[0U];
+
+         // Manually count the number of base-10 digits on the zero'th limb.
```

Comment:
This is the bit that caused me to assume that correct round was next to impossible to do efficiently in this type, but maybe it's not so bad?  I can't help but wonder if there are more efficient implementations - maybe binary search comparisons?  Or can we even do table lookup based on the most-significant-bit?  Just thinking out loud here....

Reply:
> the bit that caused me to assume that correct round was next to impossible to do efficiently in this type

Well, it sure is proving to be unruly. But there are some new efficiency provisions.
- There was an existing binary search which had calculated the number of base-10 digits in the leading limb. This is used instead of a manual counting loop now.
- For the power-of-10, a special non-`constexpr` table lookup has also been created for quick calculations of the power(s) of ten in similar matters.